src/Repository.ts

```diff
@@ -248,9 +248,11 @@ export class MongoRepository<DOC, DTO = DOC> {
   async deleteOne(conditions: any): Promise<DeleteWriteOpResultObject> {
     const collection = await this.collection;
 
-    await this.invokeEvents(PRE_KEY, [RepoOperation.delete, RepoOperation.deleteOne], conditions);
+    let document = await collection.findOne(conditions);
```

Comment:
This circumvents the `find` events, not sure we want to do that as there could be transformations happening on the document

Reply:
Was starting to retract from this, but looking at save/update it using the `this.find*` to get the original document, which does make for a strange behavior of `newDocument` being the untouched mongo document and `oldDocument` being transformed by any find events.

Reply:
Yeah, event arguments are pretty inconsistent. findOneAndUpdate passes req.updates as newDocument in Before, for example. It may be worth an overhaul and to reduce <any> usage in the next release.